From ed3f11ba1d8e81aaf93b3e4eb7c419cd70ffa7b1 Mon Sep 17 00:00:00 2001
From: Ilya Krjukov <ilya.krjukov@intel.com>
Date: Mon, 18 Mar 2013 15:23:18 +0400
Subject: Added start address and range size to JIT API debug output for VTune

BZ: 93780

This modification helps vtune developers detect problems with JIT API.
Source code tested on redhookbay device. Everything builds and works.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG; AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I0174012db50788834f70aa29c8019e6dac350cde
Orig-MCG-Change-Id: I31e9d0bfff01b9d24f8156a11fbd9d7b49534d29
Signed-off-by: Ilya Krjukov <ilya.krjukov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/Frontend.cpp          |    4 ++--
 vm/compiler/codegen/x86/Lower.cpp |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/vm/compiler/Frontend.cpp b/vm/compiler/Frontend.cpp
index 5180265..9618f93 100644
--- a/vm/compiler/Frontend.cpp
+++ b/vm/compiler/Frontend.cpp
@@ -2838,9 +2838,9 @@ static void SendTraceInfoToVTune(CompilationUnit *cUnit, JitTraceDescription *de
     int res = iJIT_NotifyEvent(iJVM_EVENT_TYPE_METHOD_LOAD_FINISHED, (void*)&jitMethod);
     if (cUnit->printMe) {
         if (res) {
-            LOGD("JIT API: a trace of '%s' method was written successfully.", cUnit->method->name);
+            LOGD("JIT API: a trace of '%s' method was written successfully address: %p, size: %d.", cUnit->method->name, jitMethod.method_load_address ,jitMethod.method_size);
         } else {
-            LOGD("JIT API: failed to write a trace of '%s' method.", cUnit->method->name);
+            LOGD("JIT API: failed to write a trace of '%s' method address: %p, size: %d.", cUnit->method->name, jitMethod.method_load_address ,jitMethod.method_size);
         }
     }
 }
diff --git a/vm/compiler/codegen/x86/Lower.cpp b/vm/compiler/codegen/x86/Lower.cpp
index a0f2d01..1cb3c28 100644
--- a/vm/compiler/codegen/x86/Lower.cpp
+++ b/vm/compiler/codegen/x86/Lower.cpp
@@ -1077,9 +1077,9 @@ void sendLabelInfoToVTune(int startStreamPtr, int endStreamPtr, const char* labe
     jitMethod.method_size = endStreamPtr-startStreamPtr;
     int res = iJIT_NotifyEvent(iJVM_EVENT_TYPE_METHOD_LOAD_FINISHED, (void*)&jitMethod);
     if (res) {
-        ALOGI("JIT API: a trace of %s method was written successfully.\n", labelName);
+        ALOGI("JIT API: a trace of %s method was written successfully address: %p size: %d.\n", labelName, jitMethod.method_load_address, jitMethod.method_size);
     } else {
-        ALOGI("JIT API: failed to write a trace of %s method.\n", labelName);
+        ALOGI("JIT API: failed to write a trace of %s method address: %p size: %d.\n", labelName, jitMethod.method_load_address, jitMethod.method_size);
     }
 }
 #endif
-- 
1.7.4.1

